java/pattern/expression/Expression.java

```diff
@@ -0,0 +1,533 @@
+/*
```

Comment:
Move back into ValueConstraint
Rename ValueConstraint.Expression to ValueConstraint.Assign